job-server/src/main/scala/spark/jobserver/JobManagerActor.scala

```diff
@@ -209,6 +214,27 @@ class JobManagerActor(contextConfig: Config, daoActor: ActorRef)
         }
       }
     }
+    case GetSparkWebUIUrl => {
+      if (jobContext.sparkContext == null) {
```

Comment:
wouldn't you want `jobContext.sparkContext.isStopped` instead? I'd prefer anything over a null check